### PR TITLE
Fix flaky integration tests and error handling for invalid team config

### PR DESCRIPTION
**Pull request by t1m0thyj**
## Proposed changes

Fix transient errors in integration tests, and upload test results in CI even when tests fail to make debugging easier.

The tests passed 10 times in a row in this workflow:
https://github.com/zowe/zowe-explorer-vscode/actions/runs/11923838338/job/33234787006

Also fixes error handling for an invalid team config file when ZE initializes, which was uncovered by fixing tests. In v3.0.3 the error notification was shown twice, and in the main branch ZE was failing to activate.

## Release Notes

Milestone: 3.1.0

Changelog: Fixed an issue during initialization where the error dialog shown for a broken team configuration file was missing the "Show Config" action

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (non-breaking change which adds or improves functionality)
- [ ] Breaking change (a change that would cause existing functionality to not work as expected)
- [ ] Documentation (Markdown, README updates)
- [ ] Other (please specify above in "Proposed changes" section)

## Checklist

**General**

- [ ] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/zowe-explorer-vscode/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] All PR dependencies have been merged and published (if applicable)
- [ ] A GIF or screenshot is included in the PR for visual changes
- [ ] The pre-publish command has been executed:
  - **v2 and below:** `yarn workspace vscode-extension-for-zowe vscode:prepublish`
  - **v3:** `pnpm --filter vscode-extension-for-zowe vscode:prepublish`

**Code coverage**

- [ ] There is coverage for the code that I have added
- [ ] I have added new test cases and they are passing
- [ ] I have manually tested the changes

**Deployment**

- [ ] I have added developer documentation (if applicable)
- [ ] Documentation should be added to Zowe Docs
  - If you're an outside contributor, please post in the [#zowe-doc Slack channel](https://openmainframeproject.slack.com/archives/CC961JYMQ) to coordinate documentation.
  - Otherwise, please check with the rest of the squad about any needed documentation before merging.
- [ ] These changes may need ported to the appropriate branches (list here):

### CODE REVIEW

**Comment by t1m0thyj** [comment]
There is another integration test (`UpdateCredentials.feature`) that has frequent transient failures, marking as draft until this one is fixed.

**Review by traeok — APPROVED** [review]
Changes make sense to me, thanks @t1m0thyj for the fix
Can re-approve if needed. The UpdateCredentials test is interesting because it tries to click on the search "action button" and Selenium times out stating that the search button isn't interactable.

**Review by traeok — DISMISSED** [review]
Thanks for adding Codicon handling to the `QuickPick` page object 😋

File: packages/zowe-explorer/.prettierignore [hunk 1]
```diff
@@ -3,4 +3,5 @@ results
 scripts
 src/webviews/dist
 l10n/bundle.l10n.json
-__tests__/__common__/.wdio-vscode-service
\ No newline at end of file
+__tests__/__common__/.wdio-vscode-service
+__tests__/__integration__/bdd/resources
```
Comment (zFernand0) [inline packages/zowe-explorer/.prettierignore:7]: thanks for ignoring the resources file 😋

**Review by zFernand0 — APPROVED** [review]

File: packages/zowe-explorer/src/utils/ProfilesUtils.ts [hunk 2]
```diff
@@ -564,7 +565,8 @@ export class ProfilesUtils {
             await ProfilesUtils.readConfigFromDisk(true);
             ZoweLogger.info(vscode.l10n.t("Zowe profiles initialized successfully."));
         } catch (err) {
-            if (err instanceof imperative.ImperativeError) {
+            // JSON parsing errors in team config files will have suppressDump=true
+            if (err instanceof imperative.ImperativeError && !err.mDetails.suppressDump) {
```
Comment (traeok) [inline packages/zowe-explorer/src/utils/ProfilesUtils.ts:569]: creative workaround for the `instanceof imperative.ImperativeError` check 😋 
I think this makes more sense than just showing the error for all imperative errors

**Review by traeok — APPROVED** [review]
Changes LGTM, thanks for adding the resources dir and for fixing these issues!

**Pull request merged**